packages/core/src/IntersectionDetector/IntersectionDetector.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useRef } from 'react'
+import { PropTypes } from '@dhis2/prop-types'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+/**
+ * @module
+ *
+ * @param {IntersectionDetector.PropTypes} props
+ * @returns {React.Component}
+ */
+export const IntersectionDetector = ({
```

Comment:
Personally I think I'd go with `VisibilityDetector` here, as most of the terminology used in this component mentions visibility. `IntersectionDetector` could be a useful name as well, but if we're going to go with `Intersection` over `Visibility` as the core responsibility of this component, I'd move all the other terminology in this component over to `Intersection` as well. Though in that case, I'd prefer just going with `IntersectionObserver`.

Reply:
I agree with @ismay that we should try to avoid be mixing terminology, and either refer to `visibility` or `intersection` consistently. I'm not 100% sure which one I would prefer here:
- `visibility` is nice, because I expect that in 99% of cases this is probably the type of logic you would want to implement with this component: when something becomes is in view, do A, when it moves out of view, do B. Unfortunately the word `visible` is also quite ambiguous and not entirely accurate. What happens with elements that have `visibility: hidden`, or `opacity: 0`, or are covered by another element? Will the `onVisibilityChange` callback be executed when these type of elements scroll into view? IMO it's not entirely clear...
- On the other hand, `intersection` is much more clear. It hints at the API being used under the hood, which can help manage the developer's expectations: it is 100% clear that this logic is based on intersections/overlaps.

🤔  On balance, my preference would probably be to settle on `IntersectionDetector` for the component and rename the prop from `onVisibilityChange` to `onIntersectionChange`. 

I also think this is probably more future proof: if we start adding support for more things the `IntersectionObserver` can do, this type of naming will make more sense.

Reply:
That's very good feedback. I've decided to go for @HendrikThePendric's suggestion.